single-line-diagram-core/src/main/java/com/powsybl/sld/layout/ZoneLayoutFactory.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.sld.layout;
+
+import com.powsybl.sld.model.ZoneGraph;
+
+/**
+ *
+ * @author Massimo Ferraro <massimo.ferraro@techrain.eu>
+ */
+public interface ZoneLayoutFactory {
+
+    ZoneLayout create(ZoneGraph graph);
```

Comment:
As for SubstationLayoutFactory we should be able to pass a SubstationLayoutFactory and a  VoltageLayoutFactory to recursively layout a zone with only automatic layouts.